core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,1092 @@
+    this._context = c;
+    this._options = o;
+    this._fullArcXY = c.isValidArc ? c.cloneLocalArc(true) : Arc3d.createUnitCircle();
+  }
+  protected get fullArcXY(): Arc3d {
+    return this._fullArcXY;
+  }
+  protected abstract createErrorProcessor(): SubdivisionErrorProcessor;
+  public computeRadiansStrictlyInsideQuadrant1(result?: number[]): number[] {
+    if (!result)
+      result = [];
+    if (this._context.isValidArc) {
+      const q = [QuadrantFractions.create(1, [0, 0.125, 0.25], true, true)];
+      const processor = this.createErrorProcessor();
+      let iters = 0;
+      do {
+        EllipticalArcApproximationContext.processQuadrantFractions(this.fullArcXY, q, processor);
+      } while (processor.isRefined && ++iters <= SubdivisionSampler.maxIters);
+      processor.getRefinedAngles(result);
+    }
+    return result;
+  }
+};
+
+/**
+ * Implementation for method `EllipticalArcSampleMethod.SubdivideForChords`
+ * @internal
+ */
+class SubdivideForChordsSampler extends SubdivisionSampler {
+  private constructor(c: EllipticalArcApproximationContext, o: EllipticalArcApproximationOptions) {
+    super(c, o);
+  }
+  public static create(context: EllipticalArcApproximationContext, options: EllipticalArcApproximationOptions): SubdivideForChordsSampler {
+    return new SubdivideForChordsSampler(context, options);
+  }
+  public override createErrorProcessor(): SubdivisionErrorProcessor {
+    return new LineStringSubdivisionErrorProcessor(this.fullArcXY, this._options.maxError);
+  }
+};
+
+/**
+ * Implementation for method `EllipticalArcSampleMethod.SubdivideForArcs`
+ * @internal
+ */
+class SubdivideForArcsSampler extends SubdivisionSampler {
+  private constructor(c: EllipticalArcApproximationContext, o: EllipticalArcApproximationOptions) {
+    super(c, o);
+  }
+  public static create(context: EllipticalArcApproximationContext, options: EllipticalArcApproximationOptions): SubdivideForArcsSampler {
+    return new SubdivideForArcsSampler(context, options);
+  }
+  public override createErrorProcessor(): SubdivisionErrorProcessor {
+    return new ArcChainSubdivisionErrorProcessor(this.fullArcXY, this._options.maxError);
+  }
+};
+
+/**
+ * Base class for processing samples of an elliptical arc.
+ * @internal
+ */
+abstract class QuadrantFractionsProcessor {
+  /**
+   * Announce the beginning of processing of the samples in a given quadrant.
+   * @return whether to process this quadrant
+  */
+  public announceQuadrantBegin(_q: QuadrantFractions): boolean { return true; }
+  /**
+   * Optionally announce a chord of the elliptical arc between the given fractions.
+   * @param pt0 start point of the chord segment
+   * @param pt1 end point of the chord segment
+   * @param f0 elliptical arc parameter in [0,1] of pt0
+   * @param f1 elliptical arc parameter in [0,1] of pt1
+   */
+  public announceChord?(pt0: Point3d, pt1: Point3d, f0: number, f1: number): void;
```

Comment:
Maybe it's too much pain to change the word `chord` but I think we use the term `segment` eveywhere else!

Reply:
Chord has additional meaning: its endpoints are on another curve (in this case, the elliptical arc). That's why I prefer it to segment.

Reply:
well, both ends are on the arc, are not they?! I mean for example, here we have 4 lines segments and all of start/end points are on the same arc, right?

![image](https://github.com/iTwin/itwinjs-core/assets/114527638/d57aa912-ef7b-4b7f-b363-69b2265a6590)

Reply:
Yes, exactly, that's why I call them chords.  More descriptive than segments.